python/mxnet/contrib/onnx/mx2onnx/_op_translations.py

```diff
@@ -219,18 +224,18 @@ def convert_fully_connected(node, **kwargs):
     """Map MXNet's FullyConnected operator attributes to onnx's Gemm operator
     and return the created node.
     """
-    onnx = import_onnx_modules()
-    name = node["name"]
-    inputs = node["inputs"]
-    attrs = node["attrs"]
+    onnx, name, proc_nodes, index_lookup, inputs, attrs = get_inputs(node, kwargs)
+
     initializer = kwargs["initializer"]
 
     no_bias = get_boolean_attribute_value(attrs, "no_bias")
 
-    input_node_id = kwargs["index_lookup"][inputs[0][0]]
-    weight_node_id = kwargs["index_lookup"][inputs[1][0]]
-
-    proc_nodes = kwargs["proc_nodes"]
+    input_node_id = index_lookup[inputs[0][0]]
```

Comment:
I am still curious why you cannot parse the inputs in line 269 using a loop of length 3(length 2 if no bias), but you instead hard coded the (input, weight, bias) explicitly?

I think it should follows a general processing function, and allow customized handling if an operator requires different handling of inputs/outputs

The current manual coding is making each operator converter super long and hard to write/modify/review/inspect.

Reply:
Agree. Submitting a new version that cleans up how inputs are retrieved.